UsbIpServer/RegistryUtils.cs

```diff
@@ -180,18 +190,45 @@ public static RegistryKey SetDeviceAsAttached(ExportedDevice device, IPAddress a
 
         public static void SetDeviceAsDetached(ExportedDevice device)
         {
-            using var deviceKey = GetDeviceKey(device, true);
-            deviceKey?.DeleteSubKeyTree(AttachedName, false);
+            using var deviceKey = GetDeviceKey(device, false);
+            if (deviceKey is null)
+            {
+                return;
+            }
+            // .NET does not have this functionality: delete a key to which you have rights while
+            // you do not have rights to the containing key. So, we must use the API directly.
+            var result = (WIN32_ERROR)PInvoke.RegDeleteKey(deviceKey.Handle, AttachedName).Value;
+            using var attached = deviceKey.OpenSubKey(AttachedName, false);
+            if (attached is not null)
+            {
+                throw new Win32Exception((int)result, "RegDeleteKey");
+            }
         }
 
         public static void SetAllDevicesAsDetached()
         {
-            using var devicesKey = GetDevicesKey(true);
+            using var devicesKey = GetDevicesKey(false);
             var deviceKeyNames = devicesKey?.GetSubKeyNames() ?? Array.Empty<string>();
+            var failure = false;
             foreach (var deviceKeyName in deviceKeyNames)
             {
-                using var deviceKey = devicesKey?.OpenSubKey(deviceKeyName, true);
-                deviceKey?.DeleteSubKeyTree(AttachedName, false);
+                using var deviceKey = devicesKey?.OpenSubKey(deviceKeyName, false);
+                if (deviceKey is null)
+                {
+                    return;
```

Comment:
Should this be `continue`? Granted, I'm not sure how we'd end up in a situation where only some of the names in `deviceKeyNames` are valid.